paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FileStoreLookupFunction.java

```diff
@@ -249,7 +250,7 @@ public Collection<RowData> lookup(RowData keyRow) {
                 rows.add(new FlinkRowData(matchedRow));
             }
             return rows;
-        } catch (OutOfRangeException e) {
+        } catch (OutOfRangeException | DimScanOverwriteException e) {
```

Comment:
Why catch DimScanOverwriteException then reopen can solve this problem.?

Reply:
Because the StreamScan doesn't read OVERWRITE snapshot (in most cases), so that dim table cannot refresh data when you overwrite the dim table. Here we just reopen the lookup table to reload the full data of dim table.

Reply:
very thanks